bungee/src/main/java/me/leoko/advancedban/bungee/BungeeMethods.java

```diff
@@ -187,7 +194,11 @@ public boolean hasPerms(Object player, String perms) {
 
     @Override
     public Permissionable getOfflinePermissionPlayer(String name) {
-        if(luckPermsSupport) return new LuckPermsOfflineUser(name);
+        if (permissionable != null) {
+            try {
+                return permissionable.getConstructor(String.class).newInstance(name);
```

Comment:
The java compiler will probably not be able to detect whether the class actually has this constructor. Using `Function<String, Permissionable> perissionableGenerator` might be a way of keeping it modular while allowing for proper code analysis by the compiler.

Reply:
Is this correct? If not, I'm sorry, I'm a beginner in Java development.

Reply:
Yes, this is perfect